pkg/services/ngalert/api/util.go

```diff
@@ -98,12 +113,21 @@ func (p *AlertingProxy) withReq(
 	newCtx, resp := replacedResponseWriter(ctx)
 	newCtx.Req = req
 
-	recipient, err := strconv.ParseInt(web.Params(ctx.Req)[":Recipient"], 10, 64)
-	if err != nil {
-		return ErrResp(http.StatusBadRequest, err, "Recipient is invalid")
-	}
+	datasourceID := web.Params(ctx.Req)[":DatasourceID"]
```

Comment:
Here since datasourceId is int64, we might want parseint instead of web.params, since the later would use 0 instead of error out when the parameter is not set.